mlflow/pytorch/__init__.py

```diff
@@ -128,6 +128,15 @@ def log_model(
     """
     Log a PyTorch model as an MLflow artifact for the current run.
 
+        .. warning::
+
+            Log the model with a signature to avoid inference errors.
+            For models without signatures, the MLflow Model Server relies on the default inferred
+            data type from NumPy. However, PyTorch often expects different defaults, particularly
+            when parsing floats. Include the signature to ensure that the model is logged with the
+            correct data type so that the MLflow model server can correctly provide valid input
```

Comment:
```suggestion
            correct data type so that the MLflow model server correctly provides valid input
```

Reply:
Just to clarify @andreakress - one thing I think would be good to emphasize is that by logging a signature, the user is making it *possible* for the model server to provide valid input. Correctly inferring the correct data types without the signature is an impossible problem. Maybe just me, but, in your suggested phrasing, it kind of feels like we're saying that there's a bug where it won't provide it correctly.

Reply:
WDYT of something like this:

If the model is logged without a signature, the MLflow Model Server relies on the default inferred data type from NumPy. However, PyTorch often expects different defaults, particularly when parsing floats. You must include the signature to ensure that the model is logged with the correct data type so that the MLflow model server can correctly provide valid input.

Reply:
Sounds good!

Reply:
Thanks for helping iterate on this one!